render/template_helpers.go

```diff
@@ -618,7 +652,49 @@ func shell(templatePath string, opts *options.BoilerplateOptions, rawArgs ...str
 	}
 
 	args, envVars := separateArgsAndEnvVars(rawArgs)
-	return util.RunShellCommandAndGetOutput(filepath.Dir(templatePath), envVars, args[0], args[1:]...)
+	workingDir := filepath.Dir(templatePath)
+	shellKey := generateShellCommandKey(args, envVars, workingDir)
+
+	// Auto-confirm all if non-interactive
+	if opts.NonInteractive {
+		opts.ShellCommandAnswers[shellKey] = true
+		util.Logger.Printf("Executing shell command (non-interactive mode)")
+		return util.RunShellCommandAndGetOutput(workingDir, envVars, args[0], args[1:]...)
```

Comment:
Do we need a length check here? What if a user only supplies a command with no args like a bash script?